src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AvoidMultipleEnumerationsTests.cs

```diff
@@ -0,0 +1,3144 @@
+        {
+            var csharpCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> h)
+    {
+        IEnumerable<int> i = Enumerable.Range(1, 10);
+        [|i|].Concat([|h|]).ToArray();
+        [|h|].Concat([|i|]).ToList();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(h As IEnumerable(Of Integer))
+            Dim i As IEnumerable(Of Integer) = Enumerable.Range(1, 10)
+            [|i|].Concat([|h|]).ToArray()
+            [|h|].Concat([|i|]).ToList()
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestTakesTwoIEnumerables2()
+        {
+            var csharpCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> h)
+    {
+        IEnumerable<int> i = Enumerable.Range(1, 10);
+        [|i|].SequenceEqual([|h|]);
+        [|h|].SequenceEqual([|i|]);
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(h As IEnumerable(Of Integer))
+            Dim i As IEnumerable(Of Integer) = Enumerable.Range(1, 10)
+            [|i|].SequenceEqual([|h|])
+            [|h|].SequenceEqual([|i|])
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestTakesTwoIEnumerables3()
+        {
+            var csharpCode = @"
+using System.Collections.Generic;
+using System.Linq;
+
+public class Bar
+{
+    public void Sub(IEnumerable<int> h)
+    {
+        IEnumerable<int> x = Enumerable.Range(1, 10);
+        [|x|].GroupJoin([|h|], i => i, i => i, (i, ints) => i).First();
+        [|x|].GroupJoin([|h|], i => i, i => i, (i, ints) => i).Last();
+    }
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csharpCode);
+
+            var vbCode = @"
+Imports System.Collections.Generic
+Imports System.Linq
+
+Namespace Ns
+    Public Class Hoo
+        Public Sub Goo(h As IEnumerable(Of Integer))
+            Dim x As IEnumerable(Of Integer) = Enumerable.Range(1, 10)
+            [|x|].GroupJoin([|h|], Function(i) i, Function(i) i, Function(i, ints) i).First()
+            [|x|].GroupJoin([|h|], Function(i) i, Function(i) i, Function(i, ints) i).Last()
+        End Sub
+    End Class
+End Namespace
+";
+            await VerifyVB.VerifyAnalyzerAsync(vbCode);
+        }
+
+        [Fact]
+        public async Task TestDelayExecutions()
+        {
+            // SkipLast and TakeLast are unavailable for NET472
+#if NETCOREAPP2_0_OR_GREATER
```

Comment:
Normally we specify the ReferenceAssemblies in the test settings instead
https://github.com/dotnet/roslyn-analyzers/blob/107d82a590a1c832f048d4f4bf1df61fd2fb3ec8/src/NetAnalyzers/UnitTests/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/UseIntegralOrStringArgumentForIndexersTests.cs#L208